code/game/machinery/ecmo.dm

```diff
@@ -0,0 +1,594 @@
+							shake_animation(4)
+							M.Weaken(3)
+							breath_mask_rip()
+							return
+						src.visible_message(SPAN_WARNING("[M] trips on \the [src]'s artery line!"), SPAN_WARNING("You trip on \the [src]'s artery line!"))
+						breath_mask = null
+						shake_animation(4)
+						M.Weaken(4)
+						update_icon()
+						return
+					src.visible_message(SPAN_WARNING("[M] barely avoids tripping on \the [src]'s artery line."), SPAN_WARNING("You barely avoid tripping on \the [src]'s artery line."))
+					shake_animation(2)
+					return
+				if(prob(25))
+					src.visible_message(SPAN_WARNING("[M] trips on \the [src]!"), SPAN_WARNING("You trip on \the [src]!"))
+					shake_animation(4)
+					M.Weaken(3)
+					return
+				src.visible_message(SPAN_WARNING("[M] almost trips on \the [src]!"), SPAN_WARNING("You almost trip on \the [src]!"))
+				shake_animation(2)
+				return
+		if(M.m_intent == M_RUN && M.a_intent == I_HURT)
+			src.visible_message(SPAN_WARNING("[M] bumps into \the [src], knocking it over!"), SPAN_WARNING("You bump into \the [src], knocking it over!"))
+			do_crash()
+	return ..()
+
+/obj/machinery/ecmo/update_icon()
+	cut_overlays()
+	if(beaker)
+		add_overlay("beaker[tipped ? "_tipped" : ""]")
+		var/datum/reagents/reagents = beaker.reagents
+		if(reagents?.total_volume)
+			var/image/filling = image('icons/obj/iv_drip.dmi', src, "[tipped ? "tipped_" : ""]reagent")
+			var/percent = round((reagents.total_volume / beaker.volume) * 100)
+			var/fill_level = 0
+			switch(percent)
+				if(-INFINITY to 9)	fill_level = 0
+				if(10 to 24)	 	fill_level = 10
+				if(25 to 49)		fill_level = 25
+				if(50 to 74)		fill_level = 50
+				if(75 to 79)		fill_level = 75
+				if(80 to 90)		fill_level = 80
+				if(91 to INFINITY)	fill_level = 100
+			filling.icon_state = "[tipped ? "tipped_" : ""]reagent[fill_level]"
+			var/reagent_color = reagents.get_color()
+			filling.icon += reagent_color
+			add_overlay(filling)
+		if(attached)
+			add_overlay("iv_in[tipped ? "_tipped" : ""]")
+
+			if(attached.get_blood_volume() < 100)
+				add_overlay("light_green[tipped ? "_tipped" : ""]")
+			if(blood_message_sent)
+				add_overlay("light_yellow[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("iv_out[tipped ? "_tipped" : ""]")
+	if(tank)
+		add_overlay("tank_oxy[tipped ? "_tipped" : ""]")
+
+		var/tank_level = 2
+		switch(tank.percent())
+			if(-INFINITY to 4)	tank_level = 0
+			if(05 to 19)		tank_level = 1
+			if(20 to 39)		tank_level = 2
+			if(40 to 59)		tank_level = 3
+			if(60 to 79)		tank_level = 4
+			if(80 to 90)		tank_level = 5
+			if(91 to INFINITY)	tank_level = 6
+		add_overlay("[tank.gauge_icon][tank_level][tipped ? "_tipped" : ""]")
+	if(breath_mask)
+		if(breather)
+			add_overlay("arterial_line_on[tipped ? "_tipped" : ""]")
+		else
+			add_overlay("arterial_line_off[tipped ? "_tipped" : ""]")
+	if(panel_open)
+		add_overlay("panel_open[tipped ? "_tipped" : ""]")
+
+/obj/machinery/ecmo/process()
```

Comment:
I am unsure as to why we have a process split here. Supposedly the ECMO should be oxygenating the blood all at once. I see this is based on the IV stand, which has separate functions to air the person and inject reagent, but this shouldn't be necessary here.

Reply:
Two cannulas are present in an ECMO, in this case only the A/V configuration is presented (Arterial/Venous), one could cannulate only with the venous line and be able to inject more blood (since the cannula is bigger), vice versa both are required for all the juicy stuff (keeping the brain up, keeping the blood and oxygenation up), as it was already in this splitted state on the IV stands, I thought to keep it as is, since it does two different things